python/generate_schema.py

```diff
@@ -17,4 +27,39 @@
         f'ArgumentParser subcommands {subcommands} != {schema_commands} in CliCommands'
     )
 
-    json.dump(cli_schema, sys.stdout, indent=2)
+    # Extract arguments for subcommands
+    # We just go straight to TypeScript here, skipping the JSONSchema intermediate.
+    ts = '/* eslint-disable */\n\n'
+    arg_map = {}
+    for subcommand, parser in choices.items():
+        ts_name = subcommand.replace('_', ' ').title().replace(' ', '') + 'Args'
+        description = parser.description
+        args = parser._get_optional_actions()
+        help = args.pop(0)
+        assert help.dest == 'help'
+        ts += f'/* {description} */\n'
+        ts += f'export interface {ts_name} {{\n'
+        for arg in args:
+            if arg.help:
+                ts += f'  /* {arg.help} */\n'
+            option_name = arg.option_strings[0].replace('--', '')
+            mod = ''
+            typ_ = 'string' if arg.type == str else 'number' if arg.type in {int, float} else 'unknown'
+            if isinstance(arg, argparse._StoreTrueAction):
```

Comment:
I don't love all this code. See my idea in the PR description about making the CLI take JSON instead.

Reply:
This gross code is gone!